client-runtime/src/main/java/com/microsoft/rest/RestProxy.java

```diff
@@ -353,7 +353,34 @@ public HttpResponse call(String responseBody) {
         return asyncResult;
     }
 
-    private Single<?> toProxyReturnValueAsync(final HttpResponse response, final SwaggerMethodParser methodParser, final Type entityType) {
+    private Single<?> handleRestResponseReturnTypeAsync(final HttpResponse response, final SwaggerMethodParser methodParser, final Type entityType) {
```

Comment:
methodParser doesn't need to be `final` anymore.

Reply:
Neither does response.

Reply:
And neither does entityType.

Reply:
Wouldn't it be nice if final were the default?